frontend/src/components/ProfileContent.tsx

```diff
@@ -50,46 +61,107 @@ const useStyles = makeStyles({
         padding: 5,
         paddingRight: 10,
     },
+    bottomRight: {
+        position: 'absolute',
+        bottom: '0',
+        right: '0',
+        transform: 'translate(-10%, -10%)',
+    },
+    bottomRight2: {
+        position: 'absolute',
+        bottom: '0',
+        right: '0',
+        transform: 'translate(-100%, -10%)',
+    },
 });
 
 export default function ProfileContent(props: ProfileContentProps) {
     const classes = useStyles();
+    const [isEditMode, toggleEditMode] = React.useState(false)
+
+    // Is used to compare fields that have been edited with their
+    // original values.
+    const [name, setName] = React.useState<string>()
+    const [username, setUsername] = React.useState<string>()
+    const [age, setAge] = React.useState<number | string>()
+    const [gender, setGender] = React.useState<string>()
+    const [occupations, setOccupations] = React.useState<string[]>()
+    const [location, setLocation] = React.useState<string>()
+    const [interests, setInterests] = React.useState<string[]>()
+    const [religion, setReligion] = React.useState<string>()
+    const [editedProfileFields, setEditableProfileFields] = React.useState<ProfileFieldList | null>(null)
```

Comment:
This seems like a good time to simplify this.

What do you think about creating a custom hook `useProfile`, which would return a profile object, and an update object. The update object could have attributes: `setName, setUsername,etc...`. In essence, the hook will just be comprised of multiple, smaller hooks